frontend/src/schemas.ts

```diff
@@ -58,24 +73,32 @@ export const RequestAddedSchema = z.object({
 
 export type RequestAdded = z.infer<typeof RequestAddedSchema>;
 
-export const RequestorErrorSchema = z.object({ error: z.literal("internal") });
+export const RequestSummarySchema = z.object({
+  id: z.number().int().gte(0),
+  method: z.string(),
+  url: z.string(),
+});
+
+export type RequestSummary = z.infer<typeof RequestSummarySchema>;
 
-export type RequestorError = z.infer<typeof RequestorErrorSchema>;
+export const RequestWithResponseSchema = z.object({
+  request: z.any(),
```

Comment:
Seems like `json-schema-to-zod` has some issues with schemas referencing other schemas, sometimes it works... and in this case it doesn't. Looking into this in another PR :)